packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -361,10 +361,42 @@ private bool ShouldGenerate(PropertyProvider property)
                 }
             }
 
-            if (customPropertyReplacements.TryGetValue(property.Name, out PropertyProvider? customProp))
+            var customPropertySerializationNames = new Dictionary<string, string>();
+            foreach (var attribute in GetCodeGenSerializationAttributes())
+            {
+                if (CodeGenAttributes.TryGetCodeGenSerializationAttributeValue(
+                    attribute,
+                    out var propertyName,
+                    out IReadOnlyList<string>? serializationNames,
+                    out _,
+                    out _,
+                    out _) && serializationNames?.Count > 0)
+                {
+                    customPropertySerializationNames.Add(propertyName, serializationNames[0]);
```

Comment:
Is this what we do in Autorest? If it doesn't make sense to have an array of serialization names, can we change the attribute?

Reply:
ah, autorest [takes](https://github.com/Azure/autorest.csharp/blob/1e93f14ea5d4dfca10c248a5e765d72363ca6800/src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs#L279) the last element from the list instead of the first. I've gone ahead and changed this property to a nullable string instead of an array since I don't see the use case for an array.